File: message_multiple/get_message_page.py
import boto3
import os
import json

import common_py.functions as f
import common_py.settings as s

def lambda_handler(event, context):
    
    resp = f.create_response_if_no_item(event, s.USER, s.LISTENER, s.PAGE_SIZE)
    if resp:
        return resp

    user = event[s.USER]
    listener = event[s.LISTENER]
    page_size = event[s.PAGE_SIZE]

    payload = json.dumps({
        s.USER: user,
        s.LISTENER: listener
    })

    client = boto3.client('lambda')
    messages = []
    for _ in range(page_size):

        response = client.invoke(
        FunctionName=os.environ['INVOKED_LAMBDA'],
        InvocationType='RequestResponse',
        LogType= 'Tail',
        Payload=payload
        )
        
        data = response['Payload'].read().decode('utf-8')
        data = json.loads(data)
        data = json.loads(data)
        print(data)

        # If it is not a dict it means that there is no message -> stop.
        if not isinstance(data[s.KEY_DATA], dict):
            break
        
        # Get only the message from the response
        data = data[s.KEY_DATA][s.KEY_MESSAGE]
        messages.append(data)
    
    if len(messages) is 0:
        return f.create_response(200, '', '', 'There are no messages')
        
    return f.create_response(200, '', messages)
 

File: message_multiple/get_message_page.js
var f = require('./common/functions')
var aws = require('aws-sdk');
aws.config.region = "eu-central-1";

exports.handler = function(e, ctx, callback) {

    let user = e.user
    let listener = e.listener
    let pageSize = e.size

    if(f.isAnyNullOrEmpty(user, listener, pageSize)) {
        callback(null, f.createResponse('',
            "Query string does not exist: from, to, listener, size OR there are empty messages", '', 400));
    } else {
        let lambda = new aws.Lambda();

        var params = {
            FunctionName: process.env.INVOKED_LAMBDA, // the lambda function we are going to invoke
            InvocationType: 'RequestResponse',
            LogType: 'Tail',
          };
          
          let messages = [];
          for(let i=0; i < pageSize; i++) {

            params.Payload = JSON.stringify({
                listener: listener,
                user: user,
            });
            
            lambda.invoke(params, function(err, data) {
                if(err) {
                    callback(null, f.createResponse('', err.message, '', 500));
                } else {
                    let parsed_data = JSON.parse(data.Payload);
                    let parsed_data2 = JSON.parse(parsed_data);
                    
                    console.log(`Message ${i}: ${parsed_data}`);
                    
                    if(parsed_data2.data !== '') 
                        messages.push(parsed_data2.data);
                   
                    if(i === pageSize - 1) {
                        if(messages.length === 0)
                            callback(null, f.createResponse('',
                                "", 'There are no queued messages', 200));
                        else
                            callback(null, f.createResponse(messages,
                                "", '', 200));
                    }
                }
            });
        }
    }
}